.editorconfig

```diff
@@ -80,3 +80,8 @@ trim_trailing_whitespace = unset
 
 [pkgs/tools/misc/timidity/timidity.cfg]
 trim_trailing_whitespace = unset
+
+# trailing whitespace is an actual syntax element of classic Markdown/
+# CommonMark to enforce a line break
+[nixos/doc/**.md]
```

Comment:
I assume this is also relevant for `md` outside `/nixos/doc`?

```diff
diff --git a/.editorconfig b/.editorconfig
index 17f92c50d16..3c7962ab4d8 100644
--- a/.editorconfig
+++ b/.editorconfig
@@ -55,6 +55,11 @@ trim_trailing_whitespace = unset
 [*.lock]
 indent_size = unset
 
+# trailing whitespace is an actual syntax element of classic Markdown/
+# CommonMark to enforce a line break
+[*.md]
+trim_trailing_whitespace = unset
+
 [eggs.nix]
 trim_trailing_whitespace = unset
 
```
Please also move it above the individual file exclusions.

Reply:
It might or might not be relevant, depending on the markdown flavor used. Github-flavored markdown for example is dropping this trailing-whitespace feature intentionally, so warnings might be appropriate there.
I do know though that the markdown variant used for nixos docs cares about trailing whitespace.

Reply:
So I am ambivalent about generalising this rule to all markdown files. It probably wouldn't do much harm if generalised, but is more precise like it is right now.

Reply:
I'm thinking that having it enforced everywhere except `nixos/doc` could be confusing, I'd much rather keep this as simple as possible.

Reply:
Fine, let's make it general then until someone using GitHub flavored markdown complains about it
¯\\_(ツ)_/¯